specification/theory_operations.adoc

```diff
@@ -1,2 +1,278 @@
+code and the `SPDM_PENDING_REQUEST (0x1)` value through the `sbiret` structure.
+The untrusted domain manager then sends the pending SPDM request to the device
+DOE mailbox. It forwards the device SPDM response to the TSM, by copying it to
+the same SPDM buffer it fetches the SPDM request from and by calling again the
+same CoVE-IO COVH call.
+This process continues until the initial CoVE-IO call is completed. The TSM then
+replies to the last COVH call with the appropriate error code and the
+`SPDM_REQUEST_COMPLETED (0x0)` value through the `sbiret` structure.
+
+The TSM only supports one pending SPDM transaction per device, and the CoVE NACL
+shared memory holds one pending SPDM action buffer per device the TSM is
+connected to. Each pending SPDM action buffer is structured as described in the
+following layout:
+
+.Pending SPDM Transaction Buffer Layout
+[options="header"]
+|===
+| Offset (bytes) | Field | Length (bytes) | Description
+
+| `0h`  | `FUNCTION_ID` | 4 | The SBI Function ID this pending SPDM transaction
+                              applies to.
+| `4h`  | `DEVICE_ID`   | 4 | The PCIe device ID this pending SPDM transaction
+                              applies to.
+| `8h`  | `SPDM_PAYLOAD_LENGTH` | 4                         | SPDM payload length.
+| `12h` | `SPDM_PAYLOAD`        | See `SPDM_PAYLOAD_LENGTH` | SPDM payload data
+                                                              (e.g. TDISP, IDE_KM).
+|===
+
+In this document, for readability reasons, flow definitions that involve SPDM
+based exchanges are simplified and do not include the above described flow
+between the TSM, the untrusted domain manager and the SPDM responder. In
+particular, the steps that cover the SPDM request generation from the TSM, the
+transmission to the host VMM, the transmission to the SPDM responder and finally
+the path back to the TSM are reduced into 2 steps:
+
+1. TSM sends SPDM request #1
+2. TSM receives SPDM response #1
+
+For example the above example would be described through those simplified steps:
+
+1. VMM calls `sbi_covh_tee_io_action()`
+2. TSM sends SPDM request #1
+3. TSM receives SPDM response #1
+4. TSM sends SPDM request #2
+5. TSM receives SPDM response #2
+6. TSM returns `sbi_covh_tee_io_action()`
+
+==== Secure SPDM Session
+
+.SPDM Session
+[source,mermaid]
+....
+%%{init: {'theme': 'neutral', 'themeVariables': {'darkMode': true}, "flowchart" : { "curve" : "basis" } } }%%
+sequenceDiagram
+
+autonumber
+
+participant VMM as Untrusted Domain Manager (VMM)
+participant TSM
+participant DSM as Device DSM or RoT
+
+VMM ->> TSM: [COVH] - sbi_covh_create_spdm_session()
+
+Note over VMM,DSM: TSM: SPDM Connection Setup
+
+TSM ->> TSM: Generate SPDM request: GET_VERSION
+TSM ->> VMM: [COVH] - spdm_req(GET_VERSION)
+VMM ->> DSM: [DOE] - SPDM_GET_VERSION
+DSM ->> VMM: [DOE] - SPDM_VERSION
+VMM ->> TSM: [COVH] - spdm_resp(VERSION)
+
+TSM ->> TSM: Generate SPDM request: GET_CAPS
```

Comment:
I recommend we use the name aligning with SPDM specification. E.g.
GET_CAPS => GET_CAPABILITIES
CAPS => CAPABILITIES
NEGOTIATE_ALGS => NEGOTIATE_ALGORITHMS
ALGS => ALGORITHMS
KEY_RESP => KEY_EXCHANGE_RSP
FINISH_RESP => FINISH_RSP

Reply:
Done, thanks.